### PR TITLE
feat(ui): add remaining bootresource actions to slice

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Added remaining bootresource actions to slice

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- N/A. Some stuff might be wrong but we'll catch it when we start working with the actions for real.

## Fixes

Fixes canonical-web-and-design/app-squad#76

### CODE REVIEW

File: ui/src/app/store/bootresource/slice.ts [hunk 1]
```diff
@@ -13,19 +19,90 @@ const bootResourceSlice = createSlice({
   initialState: {
     connectionError: false,
     eventErrors: [],
+    fetchedImages: null,
     otherImages: [],
     rackImportRunning: false,
     regionImportRunning: false,
     resources: [],
     statuses: {
-      poll: false,
```
Comment (huwshimi) [inline ui/src/app/store/bootresource/slice.ts:21]: This is totes fine, I just thought we discussed making these status names the same as the action, but maybe that was more musing that deciding.

File: ui/src/app/store/bootresource/slice.ts [hunk 2]
```diff
@@ -66,8 +143,118 @@ const bootResourceSlice = createSlice({
       state.ubuntu = payload.ubuntu;
       state.ubuntuCoreImages = payload.ubuntu_core_images;
     },
-    cleanup(state: BootResourceState) {
```
Comment (huwshimi) [inline ui/src/app/store/bootresource/slice.ts:69]: Thanks!

File: ui/src/app/store/bootresource/types/actions.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export type SaveUbuntuCoreParams = {
+  images: string[];
+};
+
+export type SaveUbuntuParams = {
+  osystems: UbuntuOsystem[];
+  source_type: BootResourceSourceType;
+};
+
+export type UbuntuOsystem = {
```
Comment (huwshimi) [inline ui/src/app/store/bootresource/types/actions.ts:28]: Should this type be before `SaveUbuntuParams`? It seems to work, but always seems odd when they're defined afterwards, but then it messes up the alphabet!

**Review by huwshimi — APPROVED** [review]
Nice! A couple of comments. I haven't validated the API calls with the MAAS code, because as you say we can improve these as needed when we're using them.

File: ui/src/app/store/bootresource/slice.ts (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/bootresource/slice.ts:21]: Oh, I thought we decided the opposite! I think this is what some would call a *complete and total breakdown of communication* hahaha. I actually like the idea of using the action name more, but maybe this is something we should update for all the models at once.

Added an issue here: https://github.com/canonical-web-and-design/maas-ui/issues/2767

File: ui/src/app/store/bootresource/types/actions.ts (hunk 3 above)
Comment (Caleb-Ellis) [reply ui/src/app/store/bootresource/types/actions.ts:28]: Yeah I'm not sure how to reconcile this. Logically if you were to read top to bottom it makes sense for `SaveUbuntuParams` to be at the top. But if you were looking for something in particular, it's easier to scan things that are alphabetical. But if you were *actually* looking for something, you'd grep it. So I dunno! Maybe it doesn't matter.

**Pull request merged**